src/creature.cpp

```diff
@@ -638,8 +638,13 @@ void Creature::onDeath()
 
 			if (attacker != this) {
 				uint64_t gainExp = getGainedExperience(attacker);
-				if (Player* player = attacker->getPlayer()) {
-					Party* party = player->getParty();
+				if (Player* attackerPlayer = attacker->getPlayer()) {
+					Player* targetPlayer = getPlayer();
+					if (targetPlayer && attackerPlayer->hasAttacked(targetPlayer)) {
+						attackerPlayer->removeAttacked(targetPlayer);
+					}
```

Comment:
See my comment below, I think those 4 lines above might be reduced to `attackerPlayer->removeAttacked(getPlayer())`, but please double check.